projects/Mallard/src/navigation/index.ts

```diff
@@ -167,13 +172,23 @@ const RootNavigator = createAppContainer(
                     }: {
                         navigation: NavigationScreenProp<{}>
                     }) => {
-                        const query = useQuery<{ hasOnboarded: boolean }>(
-                            ONBOARDING_QUERY,
-                        )
+                        const query = useQuery<{
+                            gdprAllowEssential: boolean
+                            gdprAllowPerformance: boolean
+                            gdprAllowFunctionality: boolean
+                        }>(ONBOARDING_QUERY)
                         useEffect(() => {
                             /** Setting is still loading, do nothing yet. */
                             if (query.loading) return
-                            if (!query.data.hasOnboarded) {
+                            const { data } = query
+                            // If any flag is unknown still, we want to onboard.
+                            // We expected people to give explicit yay/nay to
+                            // each GDPR bucket.
+                            if (
```

Comment:
I think just for future this might be good as a `hasFinishedOnboarding` helper that we can update test 🤷‍♂